### DISCUSSION TITLE
Druid integration: Adding an order by clause to a query might drive to different query results

### DISCUSSION THREAD

**Discussion by feralberca** [apache/superset#18932] [Ideas]
We are experiencing some issues when querying Druid using Superset. It seems that Superset is trying to optimize the query based on its content (group by, order by) which cause to use different Druid apis for solving the query (timeseries vs topN). 
As an example if I have a query with a group by clause and add some aggregation like count(), adding an order by on a field that I'm using for grouping should not change the query outcome which is what is happening now. Instead, if I query Druid directly I always get the same results with or without the order by.

**Comment by srinify** [comment]
Moving to Discussions since this isn't a bug :)

**Comment by gl** [comment]
Well that's actually an interesting question. By default, when drawing a simple timeseries line chart on one metric, Superset will add an "ORDER BY metric" that does not achieve any actual goal, as anyway records are displayed by time.

However with this ORDER BY (which you can't get rid of from the Superset UI), Druid translates the request to a "TopN" query, an approximate one, but without it it would do a "Timeseries" query, that is vectorized and accurate.

I'm not sure that handing over the ORDER BY to the query engine is very appropriate when there is no join/nesting as it changes the behaviour of the database engine.
Unless, obviously, it's done on purpose and I did not find the explanation anywhere :)
